flow360/component/simulation/models/volume_models.py

```diff
@@ -0,0 +1,128 @@
+from typing import Optional
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.models.material import Air, MaterialType
+from flow360.component.simulation.models.solver_numerics import (
+    NavierStokesSolver,
+    TransitionModelSolver,
+    TurbulenceModelSolverType,
+)
+
+
+class VolumeModelBase(Flow360BaseModel):
```

Comment:
Maybe instead have 
```
class VolumeModelBase --> volumes: EntityList[Union[GenericVolume, str]] = pd.Field()
class VolumePhysicsBase(VolumeModelBase) --> material: MaterialType = pd.Field(); nitial_conditions: Optional[dict] = pd.Field(None)
```
so you write preprocess once for expansion, also the naming for VolumeModelBase and FluidDynamicsAddOnBase is kindof awkward (FluidDynamicsAddOnBase and FluidDynamics).

Reply:
What does separating VolumeModelBase and VolumePhysicsBase do?
What's wrong the naming? FluidDynamicsAddOnBase looks unnecessary though, I can just remove it

Reply:
Yes let's then remove FluidDynamicsAddOnBase.

Reply:
The expansion logic then only exist in VolumeModelBase and you write it once. All models that inherit from the VolumeModelBase will then expand automatically. Although this probably is already included/handeled in the "preprocess/to_solver" mechanism. Ignore this if true.

The naming of `FluidDynamicsAddOnBase` and `VolumeModelBase` make it seem/sounds like the `VolumeModelBase` is the universal and general one while `FluidDynamicsAddOnBase` is some specialization of `VolumeModelBase` by adding one some more field. The FluidDynamics also coincide with the FluidDynamics model name but the FluidDynamics does not use this addon class.

I name it `VolumeModelBase` and `VolumePhysicsBase` because the reason we have `material` and `initial_condition` is because we are now in a different physics and therefore non-tightly coupled zone in your word.

Yes you can remove the `FluidDynamicsAddOnBase`. We will then need to rename the `VolumeModelBase` as it is not intuitive why some of the volume models inherit from `VolumeModelBase` but some does not.

Reply:
Renamed `VolumeModelBase` to `PDEModelBase`